cmd/bucket-handlers.go

```diff
@@ -448,11 +448,22 @@ func (api objectAPIHandlers) PostPolicyBucketHandler(w http.ResponseWriter, r *h
 	w.Header().Set("ETag", "\""+objInfo.MD5Sum+"\"")
 	w.Header().Set("Location", getObjectLocation(bucket, object))
 
-	// Set common headers.
-	setCommonHeaders(w)
+	// Decide what http response to send depending on success_action_status parameter
+	switch formValues["Success_action_status"] {
+	case "201":
+		resp := encodeResponse(PostResponse{
```

Comment:
Don't use unkeyed values place the fields properly .Since bucket and object are of string type it would be hard if they get changed and we wouldn't know. Subtle bugs like this can arise.